examples/PID_RelayOutput/PID_RelayOutput.ino

```diff
@@ -57,11 +60,8 @@ void loop()
   /************************************************
      turn the output pin on/off based on pid output
    ************************************************/
-  if (millis() - windowStartTime >= WindowSize)
-  { //time to shift the Relay Window
-    windowStartTime += WindowSize;
-    myPID.Compute();
-  }
+  myPID.Compute(); // Just update once in a window time.
+ 
   if (((unsigned int)Output > minWindow) && ((unsigned int)Output > (millis() - windowStartTime))) digitalWrite(RELAY_PIN, HIGH);
   else digitalWrite(RELAY_PIN, LOW);
 }
```

Comment:
If I may chime in...

I think it is a good ideia, but as Dlloydev pointed out: `windowStartTime` is not updated, so it breaks the logic of regularly timed relay switching. Basically, that time test will only be false when the millis counter overflows, once in every 50 days or so.

@SinisterRj, a solution that would also simplify the code is to use the return value of `Compute()` in place of the time test, since the function returns true when there is computation and false otherwise.

This is just an ideia, I have not tested it.

```suggestion
  // Returns true only once in a time window, when the output is computed.
  bool outputUpdated = myPID.Compute();

  if (outputUpdated) {
    if ((unsigned int)Output > minWindow) digitalWrite(RELAY_PIN, HIGH);
    else digitalWrite(RELAY_PIN, LOW);
  }
```

Reply:
> If I may chime in...
> 
> I think it is a good ideia, but as Dlloydev pointed out: `windowStartTime` is not updated, so it breaks the logic of regularly timed relay switching. Basically, that time test will only be false when the millis counter overflows, once in every 50 days or so.
> 
> @SinisterRj, a solution that would also simplify the code is to use the return value of `Compute()` in place of the time test, since the function returns true when there is computation and false otherwise.
> 
> This is just an ideia, I have not tested it.

@guilhermgonzaga I think that you missed a important point here: the PID output cannot be update faster than window time, but the physical arduino digital output must be updated. 
In this case, the PID output will result in a fraction of the window time (0-100%) and the digital output must be HIGH for that period of time. 
i.e.: If your window time is 5000 ms and your PID calculates an output of 3000 (60%) your digital output must be updated to be HIGH from the start of the window up to 3000 ms, and be LOW from 30001 to 5000 ms.   So your digital output must be updated every cycle.
Try this code and i think you going to have a better understanding of what I mean:
[https://github.com/Dlloydev/QuickPID/issues/41#issuecomment-1017478493](https://github.com/Dlloydev/QuickPID/issues/41#issuecomment-1017478493)

Reply:
OMG, how did I miss that?!

Sorry, my mistake.